linera-chain/src/certificate.rs

```diff
@@ -255,15 +267,25 @@ impl<T: Clone> Clone for Hashed<T> {
 }
 
 impl<T> GenericCertificate<T> {
-    pub fn new(
-        value: T,
-        old_hash: CryptoHash,
+    pub fn unchecked_new(
+        value: Hashed<T>,
         round: Round,
-        mut signatures: Vec<(ValidatorName, Signature)>,
+        signatures: Vec<(ValidatorName, Signature)>,
     ) -> Self {
+        Self {
+            value,
+            round,
+            signatures,
+        }
+    }
+
+    pub fn new(value: T, round: Round, mut signatures: Vec<(ValidatorName, Signature)>) -> Self
```

Comment:
(Are we using this at all? I would assume that if we're processing _signatures_ we always also know the hash of the value?)

Reply:
Renamed the `unchecked_new` to `new` and use it everywhere.